tests/test_picking_batch.py

```diff
@@ -305,3 +307,161 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+
+    def test20_unpickable_item_invalid_state_done(self):
+        """
+        Tests that a ValidationError is raised if the picking is on a state of
+        done
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.state = 'done'
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+        with self.assertRaises(ValidationError):
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+
+    def test18_unpickable_item_no_package_vaidation_error(self):
+        """
+        Tests that ValidationError is raised if the move_line does not have
+        a package.  This functionality is not yet handled by the system.
+        :return:
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        reason = 'missing item'
+        move_line = picking.move_line_ids[0]
+        # Make the move_line not have a package.
+        move_line.package_id = None
+        with self.assertRaises(ValidationError):
+            batch.unpickable_item(move_line_id=move_line.id,
+                                  reason=reason,
+                                  picking_type_id=None)
+
+    def test19_unpickable_item_multiple_move_lines(self):
+        """
+        Tests that a backorder is created if there are multiple move lines
+        on the picking
+        """
+        Picking = self.env['stock.picking']
+        PickingType = self.env['stock.picking.type']
+        picking_type = PickingType.search([('name', '=', 'Internal Transfer')])
+        picking, batch = self._create_valid_batch_for_location_tests()
+
+        unpickable_move_line = picking.move_line_ids[0]
+        bananas_pack = {'product': self.banana,
+                        'picking': picking,
+                        'qty': 7}
+        # Create another move_line_id that is still pickable
+        move = self.create_move(**bananas_pack)
+        self.create_move_line(move=move,
+                              qty=7,
+                              picking_id=picking.id)
+        reason = 'missing item'
+
+        batch.unpickable_item(move_line_id=unpickable_move_line.id,
+                              reason=reason,
+                              # We use a picking type here so it's easier to
+                              # assert that the move_line was created on a new
+                              # picking
+                              picking_type_id=picking_type.id)
+        # Because there are other move_line_ids that are still pickable we
+        # need to ensure that the original picking is not cancelled
+        investigation_picking = Picking.search(
```

Comment:
perhaps use backorder_id to make sure your getting the correct picking?

Reply:
Could you provide more information on how to get the backorder id?  I.E. what is it an attribute of?

Reply:
When you create a backorder it sets backorder_id to be that of the original picking.